app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -258,6 +322,14 @@ public void updateListAfterDelete(int pos) {
                 _user.getId(),
                 habitToDelete,
                 pos);
+
+        // This iterates through the habits to update their sort indexes
```

Comment:
This is the iterator to decrement all habits that have a sort index greater than the habit being deleted.